### ISSUE TITLE
Custom Processor in Build not working

### ISSUE THREAD

**Issue by AFWork1** [Unity-Technologies/InputSystem#1085]
I followed the documentation to add my own Processor, in the Editor it works just fine, if I make a build I get an error saying 
"InvalidOperationException: No processor with name 'AdjustPedalValue' (mentioned in AdjustPedalValue') has been registered"

I've tried adding it at different places and looking at similar issues I couldnt find a fix for this.

`[assembly: AlwaysLinkAssembly]
#if UNITY_EDITOR
[InitializeOnLoad]
#endif
public class AdjustPedalValue : InputProcessor<float>
{ 
#if UNITY_EDITOR
    static AdjustPedalValue()
    {
        Initialize();
    }
#endif

    [RuntimeInitializeOnLoadMethod]
    static void Initialize()
    {
        InputSystem.RegisterProcessor<AdjustPedalValue>();
    }

    public override float Process( float value, InputControl control )
    {
        return ( value * -0.5f ) + 0.5f;
    }
}
`

**Comment by stefanunity** [comment]
Could you be so nice to submit a bug report with your repro project attached through the editor. 
See https://unity3d.com/unity/qa/bug-reporting
Thanks!

**Comment by Rene-Damm** [comment]
What's the managed stripping level set to? If managed stripping is enabled, the class will need a `[Preserve]` attribute to prevent the linker from stripping it away.

**Comment by AFWork1** [comment]
It's not enabled, but to make sure I tried building with Preserve, didn't change anyhing. 
I'll try to send a bug report with a mostly empty project tomorrow if I get it there too, can't really attach the project im trying to use it with sadly. But I havent really changed anything big

**Comment by stefanunity** [comment]
Hello,

we're going to close all the github issues and will concentrate bug information in the Unity public issue tracker https://issuetracker.unity3d.com/product/unity/issues?project=116&status=1&unity_version=&view=hottest.

If you are still experienceing this issue in the latest package version please do file a bug report as described here https://unity3d.com/unity/qa/bug-reporting
For general feedback, questions and discussions you can go to https://forum.unity.com/forums/new-input-system.103/

Thank you!

**Comment by hsandt** [comment]
I have submitted the issue with a small example project. It is not public yet, but if it becomes so, I'll add the link to it here.

Note that the Custom Processor is eventually registered and should be working, so this is more of a cosmetic issue.

The most likely reason I have found is that:
- in Editor, we use [InitializeOnLoad] to call Initialize and register the processor, which is done well before Playing the game (else we wouldn't even see it in the Input Processors list when editing inputs)
- in Build, we use [RuntimeInitializeOnLoadMethod] to call Initialize on game start. But that is done a bit too late, after Input Asset loading (hence the error). But eventually it is indeed registered, so it works in the game.

It's unfortunate that issues on an open source package are solved privately, since the community may be able to bring extra information and discuss source code (also, public issues avoid sending duplicates), but I see why the Bug Reporter helps by encouraging providing a MWE. Hopefully, my bug report will become public soon anyway.

**Comment by FinlayMac** [comment]
I am also having the same issue. Following the documentation, I created a custom processor which works fine in the editor. The problem comes after exporting. 

Both the .exe and webGL versions will not register the input if the custom processor has been applied. If I remove the custom processor and reexport, the input will be registered like normal. If I use a built in processor such as Normalize, the input is processed. 

I added a Debug to my custom processor which tells me if its running. Once again, this showed it running in the editor but not the builds. I have also tried the Preserve attribute with no change.

Like @AFNuro, the error I am getting is:
InvalidOperationException: No processor with name 'MousePosRelativeToCentre' (mentioned in 'MousePosRelativeToCentre') has been registered
